src/middlewares/validation.ts

```diff
@@ -146,4 +149,44 @@ const verifyUserCredentials = async (
     }
 };
 
-export { validation, isUserExist, isAccountVerified, verifyUserCredentials, isUsersExist };
\ No newline at end of file
+const isProductExist = async(req: any, res: Response, next: NextFunction) => {
+    try {
+        const shop = await productRepositories.findItemByAttributes(Shops,"userId", req.user.id);
+        if(!shop){
+            return res.status(httpStatus.NOT_FOUND).json({ status: httpStatus.NOT_FOUND, message: "Not shop found." });
+        }
+        const isProductAvailable = await productRepositories.findByModelAndAttributes(Products,"name","shopId", req.body.name,shop.id);

```

Comment:
Let us be specific in the naming of the function so that anyone can understand the function name easily.